SpaceCore/TileSheetExtensions.cs

```diff
@@ -171,14 +173,19 @@ internal static void UpdateReferences()
                 {
                     Log.Error("WHAT? null " + asset.Key);
                     TileSheetExtensions.ExtendedTextures.Remove(oldTexture);
-                    oldTexture.Dispose();
+                    if (disposeOld)
+                        SpaceCore.TextureDisposalQueue.Enqueue(
+                            new KeyValuePair<System.IDisposable, int>(oldTexture, SpaceCore.SecondsSinceStart + TEXTURE_DISPOSAL_DELAY)
+                        );
                 }
                 else
                 {
                     TileSheetExtensions.ExtendedTextures[asset.Value.BaseTileSheet] = asset.Value;
-                    if (oldTexture != asset.Value.BaseTileSheet)
-                        oldTexture.Dispose();
-                }
+                    if (disposeOld && oldTexture != asset.Value.BaseTileSheet)
+                        SpaceCore.TextureDisposalQueue.Enqueue(
+                            new KeyValuePair<System.IDisposable, int>(oldTexture, SpaceCore.SecondsSinceStart + TEXTURE_DISPOSAL_DELAY)
+                        );
+                    }
```

Comment:
Rather than immediate disposal, we schedule it several seconds in the future

Reply:
(This is related to the other conversation below, so I'll comment there to keep the discussion in one place.)